apps/explorer/components/Entity/EntityListItem.tsx

```diff
@@ -147,3 +165,64 @@ function isValidUrl(url?: string) {
     return false
   }
 }
+
+export function getExplorerV2TxPreviewBadge(
+  tx: ExplorerV2Transaction
+): React.ReactNode {
+  // This won't work for resolution types, returning undefined.
+  let txType = getV2TransactionType(tx as V2Transaction)
+
+  // Handle the above comment.
+  if (!txType) {
+    if (tx.fileContractResolutions?.[0]) {
+      txType =
+        tx.fileContractResolutions?.[0].type === 'expiration'
+          ? 'contractExpiration'
+          : tx.fileContractResolutions?.[0].type === 'renewal'
+          ? 'contractRenewal'
+          : 'storageProof'
+    }
+  }
+
+  if (txType === 'unknown') {
+    if (tx.arbitraryData?.length)
+      return <Badge variant="simple">arbitrary data</Badge>
+  }
+
+  if (txType === 'siacoin' || txType === 'siafund')
+    return (
+      <Badge variant="gray" className="flex gap-1">
+        <ArrowLeft16 />{' '}
+        <Text color="subtle">
+          {txType === 'siacoin' ? 'siacoin' : 'siafund'}
```

Comment:
getTxTypeLabel has labels for these types, 'siacoin transfer' 'siafund transfer'